public/components/event_analytics/explorer/explorer.tsx

```diff
@@ -1430,25 +1437,17 @@ export const Explorer = ({
     [tempQuery]
   );
 
-  const generateViewQuery = (query: string) => {
-    if (query.includes(appBaseQuery)) {
-      if (query.includes('|')) {
+  const generateViewQuery = (queryString: string) => {
+    if (queryString.includes(appBaseQuery)) {
+      if (queryString.includes('|')) {
         // Some scenarios have ' | ' after base query and some have '| '
-        return query.replace(' | ', '| ').replace(appBaseQuery + '| ', '');
+        return queryString.replace(' | ', '| ').replace(appBaseQuery + '| ', '');
       }
       return '';
     }
-    return query;
+    return queryString;
   };
 
-  useEffect(() => {
-    if (isEqual(selectedContentTabId, TAB_CHART_ID)) {
-      const statsTokens = queryManager.queryParser().parse(tempQuery).getStats();
-      const updatedDataConfig = getDefaultVisConfig(statsTokens);
-      setSpanValue(!isEqual(typeof updatedDataConfig.span, 'undefined'));
-    }
-  }, [tempQuery, selectedContentTabId, curVisId]);
-
```

Comment:
why removed?

Reply:
The way metric uses to determine disabled/enabled saving-as-metric, as far as I can see, only cares about if the query contains timespan on saving-panel open. This code piece does the work to record timespan but it triggers query manager parsing and setSpanValue on every query typing/changing which causes performance degradation.